Installation issue: Flex with gettext

### Steps to reproduce the issue

```console
$ spack install flex # @2.6.4
```

Causes:

```
==> Installing flex
==> Searching for binary cache of flex
==> No binary for flex found: installing from source
==> Fetching https://github.com/westes/flex/releases/download/v2.6.4/flex-2.6.4.tar.gz
==> Fetching https://github.com/westes/flex/commit/24fd0551333e7eded87b64dd36062da3df2f6380.patch
==> Staging archive: /usr/local/var/spack/stage/flex-2.6.4-wybwdtmqdlcpo5fkbun55tatbbdxp652/flex-2.6.4.tar.gz
==> Created stage in /usr/local/var/spack/stage/flex-2.6.4-wybwdtmqdlcpo5fkbun55tatbbdxp652
==> Applied patch https://github.com/westes/flex/commit/24fd0551333e7eded87b64dd36062da3df2f6380.patch
==> Building flex [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j28'

2 errors found in build log:
     396    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-QWa6O
            4/flex-2.6.4/examples'
     397    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-QWa6O
            4/flex-2.6.4/examples'
     398    Making all in po
     399    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-QWa6
            O4/flex-2.6.4/po'
     400    *** error: gettext infrastructure mismatch: using a Makefile.in.in 
            from gettext version 0.18 but the autoconf macros are from gettext 
            version 0.19
     401    Makefile:201: recipe for target 'check-macro-version' failed
  >> 402    make[1]: *** [check-macro-version] Error 1
     403    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-QWa6O
            4/flex-2.6.4/po'
     404    Makefile:528: recipe for target 'all-recursive' failed
  >> 405    make: *** [all-recursive] Error 1
```

### Platform and user environment

Please report your OS here:
```commandline
# Docker (nvidia/cuda:9.0-base)
https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/share/picongpu/dockerfiles/ubuntu-1604/Dockerfile

$ lsb_release -d
Description:	Ubuntu 16.04 LTS
``` 
and, if relevant, post or attach:

- `packages.yaml`
```
packages:
  cuda:
    paths:
      cuda@9.0.176%gcc@5.4.0 arch=linux-ubuntu16-x86_64: /usr/local/cuda
    buildable: False
  pkg-config:
    paths:
      pkg-config@0.29.1%gcc@5.4.0 arch=linux-ubuntu16-x86_64: /usr
    buildable: False
  python:
    paths:
      python@2.7.12%gcc@5.4.0 arch=linux-ubuntu16-x86_64: /usr
    buildable: False
  openmpi:
    version: [3.1.3]
    variants: +cuda fabrics=verbs,ucx,libfabric
  hwloc:
    variants: +cuda
  all:
    providers:
      mpi: [openmpi]
```

--- a/var/spack/repos/builtin/packages/flex/package.py
+++ b/var/spack/repos/builtin/packages/flex/package.py
@@ -43,6 +43,7 @@ class Flex(AutotoolsPackage):
     # - https://github.com/westes/flex/issues/241
     patch('https://github.com/westes/flex/commit/24fd0551333e7eded87b64dd36062da3df2f6380.patch', sha256='09c22e5c6fef327d3e48eb23f0d610dcd3a35ab9207f12e0f875701c677978d3', when='@2.6.4')
 
+    @when('@:2.6.0,2.6.4')
     def autoreconf(self, spec, prefix):
         autogen = Executable('./autogen.sh')
         autogen()

--- a/var/spack/repos/builtin/packages/flex/package.py
+++ b/var/spack/repos/builtin/packages/flex/package.py
@@ -43,6 +43,10 @@ class Flex(AutotoolsPackage):
     # - https://github.com/westes/flex/issues/241
     patch('https://github.com/westes/flex/commit/24fd0551333e7eded87b64dd36062da3df2f6380.patch', sha256='09c22e5c6fef327d3e48eb23f0d610dcd3a35ab9207f12e0f875701c677978d3', when='@2.6.4')
 
+    def autoreconf(self, spec, prefix):
+        autogen = Executable('./autogen.sh')
+        autogen()
+
     @property
     def force_autoreconf(self):
         # The patch for 2.6.4 touches configure

